src/btree/bt_vrfy.c

```diff
@@ -339,6 +341,104 @@ __verify_addr_ts(WT_SESSION_IMPL *session, WT_REF *ref, WT_CELL_UNPACK *unpack,
     return (0);
 }
 
+/*
+ * __verify_history_key_with_table --
+ *     With a given history store key, and a table name, iterate through the table and check if the
+ *     key exists in the table.
+ */
+static int
+__verify_history_key_with_table(WT_SESSION_IMPL *session, const char *uri, WT_ITEM *hs_key)
+{
+    WT_CURSOR *data_cursor;
+    WT_DECL_RET;
+    WT_ITEM data_key;
+
+    WT_ERR(__wt_open_cursor(session, uri, NULL, NULL, &data_cursor));
+
+    while ((ret = data_cursor->next(data_cursor)) == 0) {
+        /* The cursor raw flag, makes sure that we can place it into a WT_ITEM */
+        F_SET(data_cursor, WT_CURSOR_RAW_OK);
+        WT_ERR(data_cursor->get_key(data_cursor, &data_key));
+        if (__wt_lex_compare(hs_key, &data_key) == 0) {
+            goto done;
+        }
+    }
+
+done:
+err:
+    return (ret);
+}
+
+/*
+ * __verify_btree_id_with_meta --
+ *     With a given btree id, iterate through the metafile table and check if the btree id exists in
+ *     any table. If exists, we place the table URI into a variable Otherwise should return
+ *     WT_NOTFOUND.
+ */
+static int
+__verify_btree_id_with_meta(WT_SESSION_IMPL *session, uint32_t btree_id, const char **uri)
+{
+    WT_CONFIG_ITEM id;
+    WT_CURSOR *meta_cursor;
+    WT_DECL_RET;
+    const char *meta_value;
+
+    WT_ERR(__wt_metadata_cursor(session, &meta_cursor));
+
+    WT_ERR(meta_cursor->reset(meta_cursor));
+    while ((ret = meta_cursor->next(meta_cursor)) == 0) {
+        meta_cursor->get_value(meta_cursor, &meta_value);
+        ret = __wt_config_getones(session, meta_value, "id", &id);
```

Comment:
I think you want to wrap this in `WT_ERR_NOTFOUND_OK`. If it's neither 0 nor `WT_NOTFOUND`, you should jump to `err` right? This code just plows right through and tries to do the `strtoul` call even if this fails.

Reply:
I ended up putting this on the next line, as there was compiler issues.